src/db/index.ts

```
@@ -279,17 +279,50 @@ export const fetchImage = async (url: string | Buffer, providerKey: string | nul
   return result
 }
 
+/**
+ * @notice Network Insertion with Retry Logic
+ * @dev Added robust error handling and retry mechanism:
+ * - Handles transaction timeouts
+ * - Implements exponential backoff
+ * - Provides detailed error reporting
+ * @param chainId The chain ID to insert
+ * @param type The network type (default: 'evm')
+ * @param t The transaction object
+ */
 export const insertNetworkFromChainId = async (chainId: types.ChainId, type = 'evm', t: Tx = db) => {
-  const [network] = await t
-    .from(tableNames.network)
-    .insert({
-      type,
-      chainId: chainId.toString(),
-    })
-    .onConflict(['networkId'])
-    .merge(['networkId'])
-    .returning<Network[]>('*')
-  return network
+  const maxRetries = 3
+  let lastError: unknown
+
+  for (let i = 0; i < maxRetries; i++) {
+    try {
+      const [network] = await t
+        .from(tableNames.network)
+        .insert({
+          type,
+          chainId: chainId.toString(),
+        })
+        .onConflict(['networkId'])
+        .merge(['networkId'])
+        .returning<Network[]>('*')
+      return network
+    } catch (err: unknown) {
```

has this ever crashed for you? i have never had it crash for me

---

I did have it fail when I had more than two concurrent providers running (set in collect/index.ts)
I added this to get an understanding of why it failed and retry.  It was silently failing before and I just didn't know why